packages/app-extensions/src/login/sagas.js

```diff
@@ -28,9 +28,13 @@ export function getOptions() {
 }
 
 export function* sessionCheck() {
-  const sessionResponse = yield call(doSessionRequest)
-  yield put(actions.setLoggedIn(sessionResponse.success))
-  if (sessionResponse.success) {
+  const {success, businessUnit} = yield call(doSessionRequest)
+  const cachedPrincipal = cache.getShortTerm('session', 'principal')
+  if (cachedPrincipal !== undefined && cachedPrincipal.currentBusinessUnit.id !== businessUnit) {
```

Comment:
IMO, type-safe checking for `undefined` often makes code more unstable instead of more stable, because it leaves out `null` which eventually also leads to a null pointer.

For this reason, I personally prefer `if (cachedPrincipal && ...` to type-safe checks for `undefined` or `null` for cases like this one. We actually don't care if `getShortTerm()` returns `undefined` or `null` here (or do we?), we only want to avoid a null pointer.